doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -323,6 +323,20 @@ at the time of use of the notation.
    scope. Obviously, expressions printed by means of such extra
    printing rules will not be reparsed to the same form.
 
+
+.. note::
+
+   When several notations can be used to print a given term, the
+   notations which capture the largest subterm of the term are used
+   preferentially. Here is an example:
```

Comment:
This note should also mention the fallback on the order in which notations were defined. (It wasn't documented so far?) And it also makes no mention of notation scopes but I suppose the scope is selected first, and then only the order is used to select between notations in the selected scope. Is that right? If yes, it should probably be written down explicitly.

Reply:
I added the extra documentation in #12984 (4b62b2f9c354c3). It seemed simpler to me than documenting here the preference to first import, then reverting this change just after in #12984.

But I can also document here and revert in #12984, if you think it is more self-contained.

Reply:
It's a bit hard to review the two PRs independently (regarding the doc). If the two should go in 8.13, I'm wondering if it wouldn't be simplest to merge this two PRs into a single one (if the main reviewers agree with this strategy of course).